src/misc/support.cpp

```diff
@@ -208,6 +208,39 @@ FILE_unique_ptr make_fopen(const char *fname, const char *mode)
 	return f ? FILE_unique_ptr(f) : nullptr;
 }
 
+static constexpr int kb_divisor = 1024L;
+static constexpr int sector_size = 512L;
+
+// File size in bytes, returns -1 on error
+int64_t stdio_size_bytes(FILE* f)
+{
+	if (!cross_fseeko(f, 0L, SEEK_END)) {
+		return cross_ftello(f);
+	}
+	return -1;
+}
+
+static int64_t stdio_size_with_divisor(FILE* f, int divisor)
+{
+	auto s = stdio_size_bytes(f);
+	if (s >= 0) {
+		return static_cast<int32_t>(s / divisor);
```

Comment:
Regarding the `int32_t` cast: given the function returns `int64_t`, this seems unnecessary?  It can do harm too, because if `divisor` is a small number like `1`, then the cast will wrap values bigger than 2 GB.

Reply:
Good point, will change. I think it was a left over from when I originally returned `int32_t`.